File: IndranilKhedkar-cs50-2017-x-pset6/credit.py
import cs50

def getDigitSum(num):
    if num < 10:
        return num;
    else:
        sum = 0
        while num != 0:
            sum = sum + num % 10;
            num = num / 10;
        return sum

def main():
    sum=0
    i=1
    while True:
        print("Number: ",end="")
        cc_number = cs50.get_int()
        if cc_number > 0:
            break

    cc_num = cc_number;

    while cc_num != 0:
        mod = cc_num%10
        if i%2 == 0:
            sum = sum + getDigitSum(mod*2)
        else:
            sum+= mod
        cc_num=cc_num/10
        i = i+1

    if (sum % 10) == 0:
        twoDigits = (cc_number / pow(10,(i-3)))
        if ((i - 1) == 15 or twoDigits==34 or twoDigits==37):
            print("AMEX");
        elif (((i - 1) == 13 or (i - 1) == 16) and twoDigits/10 == 4):
            print("VISA")
        elif ((i - 1) == 16 and (twoDigits == 51 or twoDigits == 52 or twoDigits == 53 or twoDigits == 54 or twoDigits == 55)):
            print("MASTERCARD")
    else:
        print("INVALID")

if __name__ == "__main__":
    main()

File: pset1/credit.c
#include<stdio.h>
#include<cs50.h>
#include<math.h>

int getDigitSum(int num)
{
    if(num<10)
        return num;
    else
        {
            int sum=0;
            while(num!=0)
            {
                sum=sum+num%10;
                num=num/10;
            }
            return sum;
        }
}

int main(void)
{
    long long cc_number, cc_num;
    long long sum=0, i=1, mod;
    do
    {
        printf("Number: ");
        cc_number= get_long_long();
    }while(cc_number <= 0);
    cc_num = cc_number;
    while(cc_num!=0)
    {
        mod = cc_num%10;
        if(i%2==0)
        {
            sum+= getDigitSum((int)mod*2);
        }
        else
        {
            sum+= mod;
        }
        cc_num=cc_num/10;
        i++;
    }
    if ((sum % 10) == 0)
    {
        int twoDigits = (int)(cc_number / pow(10,(i-3)));
        if ((i - 1) == 15 || twoDigits==34 || twoDigits==37)
            printf("AMEX\n");
        else if (((i - 1) == 13 || (i - 1) == 16) && twoDigits/10 == 4)
            printf("VISA\n");
        else if ((i - 1) == 16 && (twoDigits == 51 || twoDigits == 52 || twoDigits == 53 || twoDigits == 54 || twoDigits == 55))
            printf("MASTERCARD\n");
    }
    else
        printf("INVALID\n");
}
